sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -448,6 +453,10 @@ case class GetMapValue(
 
   override def toString: String = s"$child[$key]"
   override def sql: String = s"${child.sql}[${key.sql}]"
+  override def name: Option[String] = key match {
+    case NonNullLiteral(s, StringType) => Some(s.toString)
+    case _ => None
```

Comment:
Oh, do we have a valid case for this?

Reply:
e.g. `col[1]` to get the map value. It's not a single `UnresolvedAttribute` (multi-part name like `a.b.c`) and is unrelated to this bug fix